classes/class-p4-master-site.php

```diff
@@ -605,6 +606,27 @@ public function view_meta_box_search( $post ) {
 		<?php
 	}
 
+	/**
+	 * Forces a user to login using Google Auth if they have a greenpeace email
+	 * 
+	 * @param WP_User $user The current user logging in.
+	 * @param String  $username The username of the user.
+	 * @param String  $password The password of the user.
+	 */
+	function enforce_google_signon( $user, $username, $password ) {
+
+
+		if ( defined( 'WP_DEBUG' ) && is_bool( WP_DEBUG ) && WP_DEBUG ) {
+			return $user;
+		}
+
+		if ( strpos( $user->data->user_email, '@greenpeace' ) !== false ) {
```

Comment:
Maybe less likely to ever occur, but this will also match unintended addresses like `someone@greenpeace_is_cool.com`. Since the ticket says "Prohibit login with username/password for user with greenpeace.org email", I suppose we shouldn't enforce this for other greenpeace domains (not sure if they're used in emails). So we could have this check if the email address ends with `@greenpeace.com`.

Reply:
We should limit that to `greenpeace.org`. This is the global id for all GP staff. Other TLDs may not even use g-suite.